src/item.cpp

```diff
@@ -2298,6 +2245,70 @@ std::string item::display_name(unsigned int quantity) const
     return string_format("%s%s%s", name.c_str(), side.c_str(), qty.c_str());
 }
 
+std::string item::get_item_tags(bool parens) const
+{
+    std::string rtn;
+    std::stringstream ret;
+    ret.str("");
+    
+    if (is_food()) {
+        if(rotten()) {
+            ret << _(parens ? " (rotten)" : " ROTTEN");
+        } else if ( is_going_bad()) {
+            ret << _(parens ? " (old)" : " OLD");
+        } else if ( is_fresh() ) {
+            ret << _(parens ? " (fresh)" : " FRESH");
+        }
+        
+        if (has_flag("HOT")) {
+            ret << _(parens ? " (hot)" : " HOT");
+            }
+        if (has_flag("COLD")) {
+            ret << _(parens ? " (cold)" : " COLD");
+            }
+    }
+
+    if (has_flag("FIT")) {
+        ret << _(parens ? " (fits)" : " FITS");
+    }
+
+    if (is_tool() && has_flag("USE_UPS")) {
+        ret << _(parens ? " (UPS)" : " UPS");
+    }
+    if (is_tool() && has_flag("RADIO_MOD")) {
+        ret << _(parens ? " (radio:" : " Radio");
+        if( has_flag( "RADIOSIGNAL_1" ) ) {
+            ret << _(parens ? "R)" : "R");
+        } else if( has_flag( "RADIOSIGNAL_2" ) ) {
+            ret << _(parens ? "B)" : "B");
+        } else if( has_flag( "RADIOSIGNAL_3" ) ) {
+            ret << _(parens ? "G)" : "G");
+        } else {
+            ret << _(parens ? "Bug" : "Err");
+        }
+    }
+
+    if(has_flag("WET"))
+        ret << _(parens ? " (wet)" : " WET");
+
+    if(has_flag("LITCIG"))
+        ret << _(parens ? " (lit)" : " LIT");
+
+    if( already_used_by_player( g->u ) ) {
+        ret << _(parens ? " (used)" : " USED" );
+    }
+
+    if( active && !is_food() && !is_corpse() && ( type->id.length() < 3 || type->id.compare( type->id.length() - 3, 3, "_on" ) != 0 ) ) {
+        // Usually the items whose ids end in "_on" have the "active" or "on" string already contained
+        // in their name, also food is active while it rots.
+        ret << _(parens ? " (active)" : " ACTIVE" );
+    }
+
+    rtn = ret.str();
+    ret.str("");
```

Comment:
Was emptying `ret` before returning to calling function; not 100% certain if it was necessary.  Removed on working copy.